src/engine/engine.ts

```diff
@@ -187,49 +176,17 @@ export default class FilterEngine {
    */
 
   public loadedLists(): string[] {
-    return this.lists.getLoaded();
+    return [...this.lists.keys()];
   }
 
   public hasList(name: string, checksum: string): boolean {
-    return this.lists.has(name, checksum);
-  }
-
-  public deleteLists(names: string[]): boolean {
-    return this.update(this.lists.delete(names));
-  }
-
-  public deleteList(name: string): boolean {
-    return this.update(this.lists.delete([name]));
-  }
-
-  public updateLists(lists: Array<{ name: string; checksum: string; list: string }>): boolean {
-    if (this.enableUpdates === false) {
-      return false;
-    }
-
-    return this.update(this.lists.update(lists));
-  }
-
-  public updateList({
-    name,
-    checksum,
-    list,
-  }: {
-    name: string;
-    checksum: string;
-    list: string;
-  }): boolean {
-    return this.updateLists([{ name, checksum, list }]);
+    return this.lists.has(name) && this.lists.get(name) === checksum;
   }
 
   /**
    * Update engine with `resources.txt` content.
    */
   public updateResources(data: string, checksum: string): boolean {
-    if (this.enableUpdates === false) {
-      return false;
-    }
-
     if (this.resources.checksum === checksum) {
```

Comment:
Maybe we can start to verify the checksum (or maybe in a separate pr).

Reply:
You mean checking that the content we got has the same checksum? I'm wondering if we need to do that now, since we compute the checksums on the backend and lists are fetched over HTTPs it should be pretty safe?